www/js/controllers.js

```diff
@@ -59,6 +59,14 @@ angular.module('starter.controllers', ['starter.services'])
                         });
       }
     });
+    var vehicles = $resource('http://bustracker.pvta.com/infopoint/rest/vehicles/getallvehicles').query({}, function(){
+      for(var i = 0; i < vehicles.length; i++){
```

Comment:
I would normally agree, but [Mozilla's for...in page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) has this to say:

<b>Array iteration and for...in</b>

>Note: for...in should not be used to iterate over an Array where the index order is important.
Array indexes are just enumerable properties with integer names and are otherwise identical to general Object properties. There is no guarantee that for...in will return the indexes in any particular order and it will return all enumerable properties, including those with non–integer names and those that are inherited.

>Because the order of iteration is implementation-dependent, iterating over an array may not visit elements in a consistent order. Therefore it is better to use a for loop with a numeric index (or Array.prototype.forEach() or the for...of loop) when iterating over arrays where the order of access is important.

Since the data type returned by Avail is an Array (<code>query</code> is the call on <code>$resource</code> that facilitates this. Changing it to <code>get</code> would fail), it is unfortunately correct. 

We could argue that the order doesn't <i>really</i> matter, but it might at some point and that would be a crappy bug to find.